python/lsst/daf/butler/core/_butlerUri/http.py

```diff
@@ -187,7 +208,7 @@ def webdavCheckFileExists(path: Union[Location, ButlerURI, str],
 
     log.debug("Checking if file exists: %s", filepath)
 
-    r = session.head(filepath)
+    r = session.head(filepath, timeout=_defaultTimeout())
```

Comment:
It's a shame that the `session` can't have a default timeout set so you wouldn't have to add it to every call.

Reply:
I agree, it really is...